packages/web/src/components/tracks-table/TracksTable.tsx

```diff
@@ -412,17 +428,57 @@ export const TracksTable = ({
     ]
   )
 
+  const renderPurchaseButton = useCallback(
+    (cellInfo: TrackCell) => {
+      const track = cellInfo.row.original
+      const { isFetchingNFTAccess, hasStreamAccess } = trackAccessMap[
+        track.track_id
+      ] ?? { isFetchingNFTAccess: false, hasStreamAccess: true }
+      const isLocked = !isFetchingNFTAccess && !hasStreamAccess
+      const isOwner = track.owner_id === userId
+      const deleted =
+        track.is_delete || track._marked_deleted || !!track.user?.is_deactivated
+      if (!isLocked || deleted || isOwner) {
+        return null
+      }
+      // note: wrapping an if with this method does type casting for track.stream_conditions
+      if (isContentUSDCPurchaseGated(track.stream_conditions)) {
+        return (
+          <Button
+            size='small'
+            color='lightGreen'
+            className={styles.purchaseButton}
+            onClick={(e) => {
+              e.stopPropagation()
+              onClickPurchase?.(track)
+            }}
+          >
+            ${track.stream_conditions.usdc_purchase.price / 100}.00
+          </Button>
+        )
+      }
```

Comment:
should we return `null` explicitly here? or is it deterministic that we return above?